core/app/models/concerns/spree/user_methods.rb

```diff
@@ -34,6 +36,14 @@ def has_spree_role?(role_in_question)
       spree_roles.any? { |role| role.name == role_in_question.to_s }
     end
 
+    def generate_api_key(role: nil)
```

Comment:
@jhawthorn that `(spree_roles.map(&:name) & Spree::Config.roles_for_auto_api_key).any?` is so great. Thanks for that. Agreed on the name. Making the changes now.